aspects/src/main/java/org/springframework/security/authorization/method/aspectj/MethodInvocationAdapter.aj

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization.method.aspectj;
```

Comment:
This seems very similar to `org.springframework.security.access.intercept.aspectj`. Is this copied here to avoid tangles? Is there a way we can avoid this duplication?

Reply:
`MethodInvocationAdapter` in `org.springframework.security.access.intercept` is being deprecated, along with the majority of the `access` package. There isn't need for it outside of `spring-security-aspects`, so I figured it was a better home.

If needed, I can deprecate the one in `access` in this same PR.